apcd-cms/src/apps/admin_regis_table/templates/edit_registration_modal.html

```diff
@@ -0,0 +1,889 @@
+                </ul>
+              </details>
+            </div>
+          </div>
+          <div class="field-wrapper emailinput required">
+            <div class="field-errors" style="display: none"></div>
+            <label for="contact_email_${contacts}_{{r.reg_id}}"> Email<span class="asterisk">*</span> </label>
+            <input
+              type="email"
+              name="contact_email_${contacts}_{{r.reg_id}}"
+              autocomplete="email"
+              required
+              class="emailinput"
+              id="contact_email_${contacts}_{{r.reg_id}}"
+            />
+          </div>
+          <div class="field-wrapper checkboxinput">
+            <div class="field-errors" style="display: none"></div>
+            <input
+              type="checkbox"
+              name="contact_notifications_${contacts}_{{r.reg_id}}"
+              class="booleanfield"
+              id="contact_notifications_${contacts}_{{r.reg_id}}"
+            />
+            <label for="contact_notifications_${contacts}_{{r.reg_id}}">
+              Select to receive system notifications
+            </label>
+            <div id="help-text-contact_notifications_${contacts}" class="help-text">
+            </div>
+          </div>
+      `;
+      btnStatus();
+    });
+
+    dropContactBtn.addEventListener("click", () => {
+      document.getElementById(`contact_header_${contacts}_{{r.reg_id}}`).style.display = 'none';
+      let contactBlock = document.getElementById(`contact_block_${contacts}_{{r.reg_id}}`);
+      contactBlock.innerHTML = '';
+      if (contacts > 1) contacts -= 1;
+      btnStatus();
+    });
+  </script>
+
+  <!-- To require at least one "types of files" choice -->
+  <script id="form-types_of_files-required" type="module">
+    const inputs = Array.from(
+      document.querySelectorAll(`
+      input[name=types_of_files_eligibility_enrollment_{{r.reg_id}}],
+      input[name=types_of_files_provider_{{r.reg_id}}],
+      input[name=types_of_files_medical_{{r.reg_id}}],
+      input[name=types_of_files_pharmacy_{{r.reg_id}}],
+      input[name=types_of_files_dental_{{r.reg_id}}]`)
+    );
+    const inputListener = e => {
+      inputs.filter(i => i !== e.target)
+            .forEach(i => (i.required = !e.target.checked));
+    };
+
+    inputs.forEach(i => i.addEventListener('input', inputListener));
+  </script>
+
+  <!-- To require at least one "entity" identifier value -->
+  <script id="form-entity_id-required" type="module">
+    const inputs = Array.from(
+      document.querySelectorAll('input[name=fein_{{r.reg_id}}], input[name=license_number_{{r.reg_id}}], input[name=naic_company_code_{{r.reg_id}}]')
+    );
+    const inputListener = e => {
+      inputs.filter(i => i !== e.target)
+            .forEach(i => (i.required = !e.target.value.length));
+    };
+
+    inputs.forEach(i => i.addEventListener('input', inputListener));
+  </script>
+</aside>
+
```

Comment:
It is not easily maintainable to have two forms so similar. Please try to create one template to share as much as feasible for both forms.

Reply:
Very good point, just pushed up changes to reflect this feedback!